client/js/lounge.js

```diff
@@ -310,10 +314,64 @@ $(function() {
 		return msg;
 	}
 
-	function buildChannelMessages(channel, messages) {
-		return messages.reduce(function(docFragment, message) {
-			docFragment.append(buildChatMessage({
-				chan: channel,
+	function updateCondensedText(condensed, addedTypes) {
+		var obj = {};
+
+		for (var i in condensedTypes) {
+			var msgType = condensedTypes[i];
+			obj[msgType] = condensed.data(msgType) || 0;
+		}
+
+		for (var k in addedTypes) {
+			var added = addedTypes[k];
+			obj[added]++;
+			condensed.data(added, obj[added]);
+		}
+
+		var text = "";
+
+		for (var j in condensedTypes) {
+			var messageType = condensedTypes[j];
+			if (obj[messageType]) {
+				text += text === "" ? "" : ", ";
+				text += obj[messageType] + " " + messageType;
+				if (messageType === "nick" || messageType === "mode") {
+					text += " change";
+				}
+				text += obj[messageType] > 1 ? "s" : "";
+			}
+		}
+		$(condensed.children(".condensed-msg")).text(text);
+	}
+
+	function appendMessage(container, chan, chanType, messageType, msg) {
+		var condensedTypesClasses = "." + condensedTypes.join(", .");
```

Comment:
Throw this inside the `if` so that you don't unnecessarily join a string every time a message arrives.